pkg/configuration/builder/viper.go

```diff
@@ -0,0 +1,154 @@
+package builder
+
+import (
+	"fmt"
+	"log"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+const defaultConfType = "yaml"
+
+// viperBuilder is a Builder to streamline Viper configuration and building.
+// implements builder.Builder.
+type viperBuilder struct {
+	vConf *viper.Viper
+	name  string
+}
+
+// NewViper initializes and returns a new ViperBuilder.
+// configDir is the reletive path for configuration directory.
+func NewViper(configDir, appName, appEnv, appRoot string) (Builder, error) {
```

Comment:
I doubt that relates to our case. quote from mentioned doc:
> The implementing package should return concrete (usually pointer or struct) types
in our case, interface and it's implementation live under same package. AOF it should return interface so that if we change interface itself, the implementation requires change and raise compile error on correct line.
other reason, if we return a viper builder to `NewConfig`, within it we are able to make changes using viper which is not forward compatible (let's say we want to move on from viper, then that requires change on 2 packages).
we can have a chat if I couldn't deliver my thoughts here.